proposals/tags-on-existing-images.md

```diff
@@ -0,0 +1,33 @@
+# Proposal: Control the pushing of tags on existing images in target registry
+
+Relevant issues: [#2360](https://github.com/GoogleContainerTools/jib/issues/2360)
+
+## Motivation
+Currently Jib pushes tags for all built images, irrelevant of whether the image is already 
+present on the target registry. 
+
+As highlighted by @loosebazooka [#2360](https://github.com/GoogleContainerTools/jib/issues/2360) 
+this can be particularly annoying if some automated orchestrator is monitoring the registry
+and releases any new tags found. This issue is especially prevalent in cases where the tags are
+auto-generated by CI systems and added with every triggered build.
+
+## Proposal
+The idea is to give the user the ability to control this functionality by means of a switch/option. 
+There might be use-cases where the current behaviour is preferable, eg, where the tags follow semantic 
+versioning, or where the tags are not automatically generated. Hence, adding a switch seems to be 
+a better solution rather than changing the current behaviour.
+
+## Proposed changes
+A new system property is to be added to control this new behaviour:
+`-Djib.skipExistingImages`
+
+A new step is created with the below signature:
+`class CheckImageStep implements Callable<BuildResult>`, where its `call()` function performs a 
```

Comment:
We will have to see, but I feel like this step should return a manifest (as currently built in `PushImageStep.makeList()`) too, so that `PushImageStep` accepts the manifest. May need to create a "struct" class to contain manifest, imageDigest, and imageId. I think we can return imageDigest and imageId as separate fields instead of putting them into a `BuildResult` struct.

But I think these kinds of details can be sorted out when we actually work on a PR. I don't want too much bike-shedding in this proposal.